parsers/examples/example_parser.py

```diff
@@ -126,44 +76,61 @@ def fetch_production(
         )
 
     res: Response = session.get(url)
-    assert (
-        res.status_code == 200
-    ), f"Exception when fetching production for {zone_key}: error when calling url={url}"
+    if not res.status_code == 200:
+        raise ParserException(
+            "example_parser.py",
+            f"Exception when fetching production error code: {res.status_code}: {res.text}",
+            zone_key,
+        )
```

Comment:
do we not want the target_datetime in the error message to be able to track when the parser fails better?

Reply:
Maybe it would be better to update the parser exception to also accept the target datetime?

Right now you have to include it as a text string in the message

Reply:
@unitrium you had a more long-term idea about this?

Reply:
I am working on an improved logger to have this.
It's outside of the scope of this PR.